cloud/deployment/deployment.go

```diff
@@ -522,46 +532,41 @@ func selectDeployment(deployments []astro.Deployment, message string) (astro.Dep
 	return selected, nil
 }
 
-func GetDeployment(ws, deploymentID string, client astro.Client) (astro.Deployment, error) {
+func GetDeployment(ws, deploymentID, deploymentName string, client astro.Client) (astro.Deployment, error) {
 	deployments, err := getDeployments(ws, client)
 	if err != nil {
 		return astro.Deployment{}, errors.Wrap(err, errInvalidDeployment.Error())
 	}
 
+	if deploymentID != "" && deploymentName != "" {
+		fmt.Printf("Both a Deployment ID and Deployment Name have been supplied. The Deploymnet ID %s will be used\n", deploymentID)
+	}
+	// find deployment by name
+	if deploymentID == "" && deploymentName != "" {
+		var stageDeployments []astro.Deployment
+		for i := range deployments {
+			if deployments[i].Label == deploymentName {
+				stageDeployments = append(stageDeployments, deployments[i])
+			}
+		}
+		if len(stageDeployments) > 1 {
+			fmt.Printf("More than one Deployment with the name %s was found\n", deploymentName)
```

Comment:
nitpick: in this case shouldn't we allow the user to select one out of these deployments only, instead of giving the whole deployments list